File: python/p004.py
def compute(LIMIT):
	palindrome = 0
	for i in range(LIMIT, 100, -1):
		j = LIMIT
		while j > 100:
			product = i * j
			if palindrome > product:
				break
			else:
				reverse = 0
				while product != 0:
					reverse = reverse * 10 + product % 10
					product //= 10
				if reverse == (i * j):
					palindrome = (i * j)
			j -= 1
	return palindrome

if __name__ == "__main__":
	print(compute(999))

File: c/p004.c
#include "stdio.h"

int compute(const int LIMIT) {
	int palindrome = 0;
	for (int i = LIMIT; i > 100; i--) {
		for (int j = LIMIT; j > 100; j--) {
			if (palindrome > (i * j))
				break;
			else {
				int product = i * j;
				int reverse = 0;
				while (product != 0) {
					reverse = reverse * 10 + product % 10;
					product /= 10;
				}
				if (reverse == (i * j))
					palindrome = i * j;
			}
		}
	}
	return palindrome;
}

int main() {
	printf("%i\n", compute(999));
	return 0;
}
